tests/test_transpilers_placer.py

```diff
@@ -0,0 +1,161 @@
+import networkx as nx
+import pytest
+from qibo import gates
+from qibo.models import Circuit
+
+from qibolab.transpilers.placer import (
+    Backpropagation,
+    Custom,
+    PlacementError,
+    Random,
+    Subgraph,
+    Trivial,
+    assert_mapping_consistency,
+    assert_placement,
+)
+
+
+def star_connectivity():
+    Q = ["q" + str(i) for i in range(5)]
+    chip = nx.Graph()
+    chip.add_nodes_from(Q)
+    graph_list = [
+        (Q[0], Q[2]),
+        (Q[1], Q[2]),
+        (Q[3], Q[2]),
+        (Q[4], Q[2]),
+    ]
+    chip.add_edges_from(graph_list)
+    return chip
+
+
+def star_circuit():
+    circuit = Circuit(5)
+    for i in range(1, 5):
+        circuit.add(gates.CNOT(i, 0))
+    return circuit
+
+
+def test_assert_placement_true():
+    layout = {"q0": 0, "q1": 1, "q2": 2, "q3": 3, "q4": 4}
+    circuit = Circuit(5)
+    assert_placement(circuit, layout)
+
+
+@pytest.mark.parametrize("qubits", [5, 3])
+@pytest.mark.parametrize("layout", [{"q0": 0, "q1": 1, "q2": 2, "q3": 3}, {"q0": 0, "q0": 1, "q2": 2}])
+def test_assert_placement_false(qubits, layout):
+    circuit = Circuit(qubits)
+    with pytest.raises(PlacementError):
+        assert_placement(circuit, layout)
+
+
+def test_mapping_consistency_true():
+    layout = {"q0": 0, "q1": 2, "q2": 1, "q3": 4, "q4": 3}
+    assert_mapping_consistency(layout)
+
+
+@pytest.mark.parametrize(
+    "layout", [{"q0": 0, "q1": 0, "q2": 1, "q3": 4, "q4": 3}, {"q0": 0, "q1": 2, "q0": 1, "q3": 4, "q4": 3}]
```

Comment:
Since this kind of placement is used many times, can maybe be useful to write a simple function to do that?
For example:

```python
def set_layout_for_given_map(target_list):
    return [{"q" + str(i) : str(target_list[i])} for i in range(len(target_list))]
```

Reply:
It is used only here...

Reply:
True, but more than ten times!

Reply:
It is alredy implemented for the custom layout, you can pass both a list or a dict. the problem is that for the custom layout it raises errors if some elements are duplicated. moreover in the tests sometimes i duplicate physical qubits ("qi") and sometimes logical qubits, so i prefer to have more control by directly defining the dictionary.